styx-cli/src/main/java/com/spotify/styx/cli/Main.java

```diff
@@ -250,265 +232,158 @@ private void run() {
           // parsing unknown command will fail so this would only catch non-exhaustive switches
           throw new ArgumentParserException("Unrecognized command: " + command, parser.parser);
       }
+      signaller.signalShutdown();
     } catch (ArgumentParserException e) {
       parser.parser.handleError(e);
       System.exit(EXIT_CODE_ARGUMENT_ERROR);
-    } catch (InterruptedException | ExecutionException e) {
+    } catch (IOException e) {
       e.printStackTrace();
       System.exit(EXIT_CODE_UNKNOWN_ERROR);
-    } catch (Exception e) {
-      System.err.println(e.getClass().getSimpleName() + ": " + e.getMessage());
-      System.exit(EXIT_CODE_API_ERROR);
     }
   }
 
-  private void backfillCreate() throws ExecutionException, InterruptedException, IOException {
+  private void backfillCreate() {
     final String component = namespace.getString(parser.backfillCreateComponent.getDest());
     final String workflow = namespace.getString(parser.backfillCreateWorkflow.getDest());
     final String start = namespace.getString(parser.backfillCreateStart.getDest());
     final String end = namespace.getString(parser.backfillCreateEnd.getDest());
     final int concurrency = namespace.getInt(parser.backfillCreateConcurrency.getDest());
 
-    final BackfillInput backfillInput = BackfillInput.create(
-        Instant.parse(start), Instant.parse(end), component, workflow, concurrency);
-
-    final ByteString payload = serialize(backfillInput);
-    final ByteString response = send(
-        Request.forUri(apiUrl("backfills"), "POST").withPayload(payload));
-    final Backfill backfill = deserialize(response, Backfill.class);
-    cliOutput.printBackfill(backfill);
+    styxClient.backfillCreate(component, workflow, start, end, concurrency)
+        .whenComplete((backfill, throwable) -> cliOutput.printBackfill(backfill));
```

Comment:
Shouldn't we check throwable as well here? Since in the exception handler we don't really use response, `CompletionStage:exceptionally` would be better?